### ISSUE TITLE
Keyboard shortcuts for next/previous tab

### ISSUE THREAD

**Issue by brianlukoff** [mingo-app/mingo#440]
From muscle memory I keep wanting to press Cmd+Shift+[ to move to the previous tab or Cmd+Shift+] to move to the next tab. Is it possible to add those keyboard shortcuts?

**Comment by brianlukoff** [comment]
Thanks!

**Comment by tothradoslav** [comment]
Hi, we have uploaded a new release candidate with this issue corrected in it. Please check it out and let me know if it works fine. Thank you for your input!

https://github.com/mingo-app/mingo/releases/tag/v1.9.1-rc1

**Comment by sagrawal31** [comment]
I have a different view on the keyboard shortcut for the next/previous tab. Since major browsers (Chrome, Safari, Firefox), hybrid & native apps (like Canva, Miro, Notion, Apple freeform and many more), and IDEs (example: IntelliJ, TextMate) use a common keyboard shortcut i.e. `Ctrl + Tab` for next tab and `Ctrl + Shift + Tab` for previous tab, it would be seamless to support those shortcuts in Mingo as well.

Would like to reopen this issue.

**Comment by tothradoslav** [comment]
CTRL + tab works for toggling the previous tab. This is how I see it working in intelliJ, for example. This is, i guess, a fairly subjective issue, as for me, the behaviour in Chrome, for example, is annoying :)

**Comment by sagrawal31** [comment]
Okay, it's an opinionated choice. Truly your decision!

**Comment by sagrawal31** [comment]
By the way, the existing shortcuts (to navigate) are still useless conditionally. Pressing ⌘LEFT or ⌘RIGHT does not work when the input field is focused either in **Document**, **Aggregation** or **NodeShell** tabs, so I have to first **blur the input fields with the mouse** and then use the shortcuts to change the tab so shortcuts are of no use because the primary use of Mingo is to query data so 70% of the times, an input field will have the focus.



https://github.com/mingo-app/mingo/assets/1804514/a079008e-e9f9-4765-a504-4b64ca732eef

**Comment by sagrawal31** [comment]
@tothradoslav @achtan, can we please reopen this issue? The navigation between the tabs is a real productivity killer.

**Comment by sagrawal31** [comment]
Thank you, @tothradoslav for quickly reopening the issue.

Any thoughts on this-

> Pressing ⌘LEFT or ⌘RIGHT does not work when the input field is focused either in Document, Aggregation or NodeShell tabs

**Comment by tothradoslav** [comment]
This is older issue, but looking at it again i have one issue with the implementation of ⌘LEFT or ⌘RIGHT in input fields: natively, this moves the cursor to the end / beginning of a row. Shall we override this? Not sure...

**Comment by sagrawal31** [comment]
Yes, that will be a problem if ⌘LEFT or ⌘RIGHT won't be able to move the cursor to start/end. My vote is not to overwrite this behaviour. 

The `Cmd+Shift+[` and `Cmd+Shift+]` is fine. Or better, if a configuration option can be provided to customise this with

> [keyboard shortcut i.e. Ctrl + Tab for next tab and Ctrl + Shift + Tab for previous tab](https://github.com/mingo-app/mingo/issues/440#issuecomment-1609410967)

**Comment by tothradoslav** [comment]
I will close this for now, not so important.
